main.go

```diff
@@ -229,58 +233,96 @@ func main() {
 
-	// set up the registerer
-	basicConfig := webhookClient.BasicConfig{
-		Timeout:         config.Webhook.Timeout,
-		RegistrationURL: config.Webhook.RegistrationURL,
-		Request: webhook.W{
-			Config: webhook.Config{
-				URL: config.Webhook.Request.WebhookConfig.URL,
+	var acquirer *acquire.RemoteBearerTokenAcquirer
+	var periodicRegisterer *webhookClient.PeriodicRegisterer
+
+	for i := 1; i <= config.Webhook.WebhookCount; i++ {
+		// set up the registerer
+		basicConfig := webhookClient.BasicConfig{
+			Timeout:         config.Webhook.Timeout,
+			RegistrationURL: config.Webhook.RegistrationURL + "?webhook=" + strconv.Itoa(i),
+			Request: webhook.W{
+				Config: webhook.Config{
+					URL: config.Webhook.Request.WebhookConfig.URL + "?webhook=" + strconv.Itoa(i),
+				},
+				Events:     []string{config.Webhook.Request.Events},
+				FailureURL: config.Webhook.Request.WebhookConfig.FailureURL + "?webhook=" + strconv.Itoa(i),
 			},
-			Events:     []string{config.Webhook.Request.Events},
-			FailureURL: config.Webhook.Request.WebhookConfig.FailureURL,
-		},
-	}
+		}
 
-	acquireConfig := acquire.RemoteBearerTokenAcquirerOptions{
-		AuthURL:        config.Webhook.JWT.AuthURL,
-		Timeout:        config.Webhook.JWT.Timeout,
-		Buffer:         config.Webhook.JWT.Buffer,
-		RequestHeaders: config.Webhook.JWT.RequestHeaders,
-	}
+		acquireConfig := acquire.RemoteBearerTokenAcquirerOptions{
+			AuthURL:        config.Webhook.JWT.AuthURL,
+			Timeout:        config.Webhook.JWT.Timeout,
+			Buffer:         config.Webhook.JWT.Buffer,
+			RequestHeaders: config.Webhook.JWT.RequestHeaders,
+		}
 
-	acquirer, err := acquire.NewRemoteBearerTokenAcquirer(acquireConfig)
-	if err != nil {
-		logging.Error(logger).Log(logging.MessageKey(), "failed to create bearer auth plain text acquirer:", logging.ErrorKey(), err.Error())
-		os.Exit(1)
-	}
+		acquirer, err = acquire.NewRemoteBearerTokenAcquirer(acquireConfig)
+		if err != nil {
+			logging.Error(logger).Log(logging.MessageKey(), "failed to create bearer auth plain text acquirer:", logging.ErrorKey(), err.Error())
+			os.Exit(1)
+		}
 
-	registerer, err := webhookClient.NewBasicRegisterer(acquirer, secretGetter, basicConfig)
-	if err != nil {
-		logging.Error(logger).Log(logging.MessageKey(), "failed to setup registerer", logging.ErrorKey(), err.Error())
-		os.Exit(1)
-	}
+		registerer, err := webhookClient.NewBasicRegisterer(acquirer, secretGetter, basicConfig)
+		if err != nil {
+			logging.Error(logger).Log(logging.MessageKey(), "failed to setup registerer", logging.ErrorKey(), err.Error())
+			os.Exit(1)
+		}
+
+		periodicRegisterer = webhookClient.NewPeriodicRegisterer(registerer, config.Webhook.RegistrationInterval, logger)
 
-	periodicRegisterer := webhookClient.NewPeriodicRegisterer(registerer, config.Webhook.RegistrationInterval, logger)
+		// start the registerer
+		periodicRegisterer.Start()
+	}
 
-	// start the registerer
-	periodicRegisterer.Start()
+	// // set up the registerer
```

Comment:
good practice is not to leave code commented that we don't need anymore; if you want to see how main() used to look you can always through the history of this file and find an earlier commit.  I would suggest deleting these commented out lines.

Reply:
My bad, I thought I had removed them before committing the changes, but I'll get rid of them.